local/experimental/blockchain.go

```diff
@@ -0,0 +1,490 @@
+	subnetIDs := make([]ids.ID, numSubnets)
+	for i := 0; i < numSubnets; i++ {
+		w.log.Info("creating subnet tx")
+		cctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+		subnetID, err := w.wallet.IssueCreateSubnetTx(
+			&secp256k1fx.OutputOwners{
+				Threshold: 1,
+				Addrs:     []ids.ShortID{w.addr},
+			},
+			common.WithContext(cctx),
+			defaultPoll,
+		)
+		cancel()
+		if err != nil {
+			return nil, err
+		}
+		w.log.Info("created subnet tx", zap.String("subnet-ID", subnetID.String()))
+		subnetIDs[i] = subnetID
+	}
+	return subnetIDs, nil
+}
+
+func (w *wallet) addPrimaryValidators(
+	ctx context.Context,
+	ln LocalNetwork,
+	names map[string]struct{},
+) error {
+	w.log.Info(logging.Green.Wrap("adding the nodes as primary network validators"))
+	// ref. https://docs.avax.network/build/avalanchego-apis/p-chain/#platformgetcurrentvalidators
+	cctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+	vs, err := w.client.GetCurrentValidators(cctx, constants.PrimaryNetworkID, nil)
+	cancel()
+	if err != nil {
+		return err
+	}
+	curValidators := make(map[ids.NodeID]struct{})
+	for _, v := range vs {
+		curValidators[v.NodeID] = struct{}{}
+	}
+	addedValidators := []ids.NodeID{}
+	for nodeName := range names {
+		// TODO: handle missing name
+		node := ln.GetNodeUnsafe(nodeName)
+		nodeID := node.GetNodeID()
+
+		_, isValidator := curValidators[nodeID]
+		if isValidator {
+			continue
+		}
+
+		// Prepare node BLS PoP
+		//
+		// It is important to note that this will ONLY register BLS signers for
+		// nodes registered AFTER genesis.
+		blsKeyBytes, err := base64.StdEncoding.DecodeString(node.GetConfig().StakingSigningKey)
+		if err != nil {
+			return err
+		}
+		blsSk, err := bls.SecretKeyFromBytes(blsKeyBytes)
+		if err != nil {
+			return err
+		}
+		proofOfPossession := signer.NewProofOfPossession(blsSk)
+
+		cctx, cancel = context.WithTimeout(ctx, defaultTimeout)
+		txID, err := w.wallet.IssueAddPermissionlessValidatorTx(
+			&txs.SubnetValidator{
+				Validator: txs.Validator{
+					NodeID: nodeID,
+					Start:  uint64(time.Now().Add(validationStartOffset).Unix()),
+					End:    uint64(time.Now().Add(validationDuration).Unix()),
+					Wght:   genesis.LocalParams.MinValidatorStake,
+				},
+				Subnet: ids.Empty,
+			},
+			proofOfPossession,
+			w.wallet.AVAXAssetID(),
+			&secp256k1fx.OutputOwners{
+				Threshold: 1,
+				Addrs:     []ids.ShortID{w.addr},
+			},
+			&secp256k1fx.OutputOwners{
+				Threshold: 1,
+				Addrs:     []ids.ShortID{w.addr},
+			},
+			10*10000, // 10% fee percent, times 10000 to make it as shares
+			common.WithContext(cctx),
+		)
+		cancel()
+		if err != nil {
+			return err
+		}
+		w.log.Info("added node as primary network validator", zap.String("name", nodeName), zap.String("nodeID", nodeID.String()), zap.String("txID", txID.String()))
+		addedValidators = append(addedValidators, nodeID)
+	}
+
+	// Ensure all added nodes are primary validators (else add subnet validator will revert)
```

Comment:
I am not sure avoid this wait, maybe the network want to do something else before the nodes are validators, this wait is about min 20 secs I think?